cpp/include/raft/neighbors/detail/refine.cuh

```diff
@@ -108,17 +108,17 @@ void refine_device(raft::device_resources const& handle,
     handle.get_thrust_policy(), fake_coarse_idx.data(), fake_coarse_idx.data() + n_queries);
 
   raft::neighbors::ivf_flat::index<data_t, idx_t> refinement_index(
-    handle, metric, n_queries, false, dim);
+    handle, metric, n_queries, false, false, dim);
```

Comment:
The index will be filled once, and then discarded. We can set conservative_memory_allocation = true.
```suggestion
    handle, metric, n_queries, false, true, dim);
```